interpreter/runtimecost.go

```diff
@@ -144,9 +146,25 @@ func NewCostTracker(estimator ActualCostEstimator, opts ...CostTrackerOption) (*
 	return tracker, nil
 }
 
+// FunctionCostTracker binds a function overload to a runtime FunctionTracker implementation.
+//
+// FunctionCostTracker instances augment or override ActualCostEstimator decisions, allowing for  versioned and/or
+// optional cost tracking changes.
+func FunctionCostTracker(function, overloadID string, fnTracker FunctionTracker) CostTrackerOption {
+	return func(tracker *CostTracker) error {
+		functionKey := fmt.Sprintf("%s|%s", function, overloadID)
+		tracker.functionTrackers[functionKey] = fnTracker
```

Comment:
What if no such `functionKey` in functionTrackers?

Reply:
@cici37 you mean later on in the code? The function trackers override or extend the base estimator / tracker implementations, so the default behavior for the base tracker would be used.

Reply:
It will panic/npe and break all existing tests while testing in k8s side..
```
if _, ok := tracker.functionTrackers[functionKey]; ok {
			tracker.functionTrackers[functionKey] = fnTracker
		}
```
solved it..

Reply:
@cici37 I don't follow what issue you're seeing. The `functionTrackers` is just a map and there's an assignment to a map key. This should always work unless the `functionTrackers` is not initialized because the `CostTracker` is initialized directly rather than with `NewCostTracker()`

Reply:
The functionTrackers is not initialized at this point. And ContextEval func is used from k8s. Thanks for addressing this!